src/components/MyAccount/RequestsTab/FreezeButton.tsx

```diff
@@ -1,15 +1,47 @@
 import { useState } from "react"
 import type { Hold, Patron } from "../../../types/myAccountTypes"
-import { Button } from "@nypl/design-system-react-components"
+import {
+  Box,
+  Button,
+  Heading,
+  Icon,
+  useModal,
+  Text,
+} from "@nypl/design-system-react-components"
+import styles from "../../../../styles/components/MyAccount.module.scss"
 
 const FreezeButton = ({ hold, patron }: { hold: Hold; patron: Patron }) => {
   const [frozen, setFrozen] = useState(hold.frozen)
   const [isDisabled, setIsDisabled] = useState(false)
+  const { onOpen: openModal, onClose: closeModal, Modal } = useModal()
+  const modalProps = {
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Text sx={{ marginLeft: "l", marginRight: "m" }}>
+          Please try again.
+        </Text>
+      </Box>
+    ),
+    closeButtonLabel: "OK",
+    headingText: (
+      <Heading className={styles.modalHeading}>
+        <>
+          <Icon size="large" name="errorFilled" color="ui.error.primary" />
+          <Text sx={{ marginBottom: 0 }}> Freezing this hold failed </Text>
+        </>
+      </Heading>
+    ),
+    onClose: () => {
+      closeModal()
+      setFrozen(!frozen)
```

Comment:
When updating a variable based on its current value, use the callback function syntax:
```
setFrozen((prevValue) => !prevValue)
```

This is mostly to cover any race conditions or rendering issues since the setState function is asynchronous.